typescript/packages/wallets/crossmint/src/balance.ts

```diff
@@ -0,0 +1,49 @@
+import type { CrossmintApiClient } from "@crossmint/common-sdk-base";
+import type { Chain, EVMWalletClient, Plugin } from "@goat-sdk/core";
+import { z } from "zod";
+import { SUPPORTED_CURRENCIES, SupportedEVMCurrency, SupportedSolanaCurrency } from "./api";
+import { CrossmintEVMSmartWalletClient } from "./smart-wallet";
+import { CrossmintSolanaCustodialWalletClient } from "./custodial";
+
+export const getBalanceParametersSchema = z.object({
+    wallet: z.string().optional().describe("The address to check the balance of"),
+    currencies: z.array(z.enum(SUPPORTED_CURRENCIES)).optional()
+        .describe("The currencies to check balances for"),
+    chains: z.array(z.string()).optional()
+        .describe("The chains to check balances for"),
+});
+
+export function balanceFactory(client: CrossmintApiClient) {
+    return function balance(): Plugin<CrossmintEVMSmartWalletClient | CrossmintSolanaCustodialWalletClient> {
+        return {
+            name: "Crossmint Balance",
+            supportsChain: (chain: Chain) => true,
+            supportsSmartWallets: () => true,
+            getTools: async () => {
+                return [
+                    {
+                        name: "get_token_balances",
+                        description: "This {{tool}} gets your wallet token balances",
```

Comment:
I'd be maybe more specific here (cross-chain, multiple tokens, etc.) 

Also it works for other wallets too not just yours! :)

Reply:
good point